src/meepgeom.cpp

```diff
@@ -1717,16 +1719,34 @@ fragment_stats::fragment_stats(geom_box &bx)
   num_pixels_in_box = get_pixels_in_box(&bx);
 }
 
+void fragment_stats::reset() {
+  resolution = 0;
+  split_chunks_evenly = false;
+  destroy_geom_box_tree(geom_tree);
+  geom_tree = NULL;
+}
+
 void fragment_stats::init_libctl(material_type default_mat, bool ensure_per, meep::grid_volume *gv,
                                  vector3 cell_size, vector3 cell_center,
                                  geometric_object_list *geom_) {
   geom_initialize();
   default_material = default_mat;
-  ensure_periodicity = ensure_per;
   geometry_center = cell_center;
   dimensions = meep::number_of_directions(gv->dim);
   geometry_lattice.size = cell_size;
   geom_fix_object_list(*geom_);
+  geom_box cell_box =  make_box_from_cell(cell_size);
+  int temp_dims = dimensions;
+  int temp_periodicity = ensure_periodicity;
+  dimensions = 3;
+  ensure_periodicity = 0;
+  geom_tree = create_geom_box_tree0(*geom_, cell_box);
```

Comment:
As a simpler alternative to creating this tree, you could also just cache `geom_get_bounding_box` for each object, and check whether the bounding box overlaps before calling `box_overlap_with_object`.  This only helps the constant factor, though, unlike a tree which in principle can give better scaling.